src/RemediationRoleSaver.cpp

```diff
@@ -92,6 +92,18 @@ void RemediationSaverBase::selectFilenameAndSaveRole()
 }
 
 
+void RemediationSaverBase::removeFileWhenEmpty(const QString& filename)
+{
+    QFile outputFile(filename);
+    outputFile.open(QIODevice::ReadOnly);
+    int fileSize = outputFile.size();
+    outputFile.close();
+
+    if (fileSize == 0)
+        QFile::remove(filename);
+}
```

Comment:
IMO we can avoid all of this, this is added complexity that the rest of the workbench doesn't do anyway.

Reply:
I got the impression that you would like to delete the empty files after a failed generation attempt.
Moreover - I think that creating empty files is unavoidable when using the library. I have to open the file, pass the descriptor to the function that does the generation, then I have to close the file regardless whether the generation went well or not - this produces either a remediation role, or an empty file.

Reply:
I think this is a little excessive, I was just using this example to demonstrate that there is no error reporting. It can probably stay though..